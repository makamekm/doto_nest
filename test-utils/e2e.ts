import { toMatchImageSnapshot } from 'jest-image-snapshot';
import 'jest-environment-puppeteer';
import config from '../nuxt.config';
import app, {tryStartBootstrap} from '../api/main';
import { Server } from 'net';
import getPort from 'get-port';

jest.setTimeout(60000);

process.env.IS_EXPRESS = 'true';

config.dev = true;
let port: number;

let server: Server;

beforeAll(async () => {
  port = await getPort();
  await tryStartBootstrap();
  await new Promise(r => {
    server = app.listen(port, r);
  });
});

afterAll(async () => {
  await new Promise(r => server.close(r));
});

expect.extend({ toMatchImageSnapshot });

export const find = async (selector) => {
  await page.waitForSelector(selector);
  return page.$(selector);
};

export const findAll = async (selector) => {
  await page.waitForSelector(selector);
  return page.$$(selector);
};

export const containsText = async (text, wrapperSelector) => {
  const textSelector = `//text()[contains(.,'${text}')]`;
  const wrapper = wrapperSelector ? await find(wrapperSelector) : page;

  await page.waitForXPath(textSelector);
  const matches = await wrapper!.$x(textSelector);

  return matches.length > 0;
};

export const open = url => page.goto(`http://localhost:${port}${url}`, { waitUntil: `networkidle0` });
