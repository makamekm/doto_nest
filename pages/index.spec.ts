import { shallowMount, mount } from '@vue/test-utils';
import page from './index.vue';
import getLocalVue from '~/utils/localVue';
import getContext from '~/utils/context';
import { Store } from 'vuex';
import VueRouter from 'vue-router';

describe('page index', () => {

  const getData = () => ([{
    id: 1,
    version: {
      id: 0,
      version: 'qwerty',
    },
    name: 'test',
    status: 'good',
    author: 'maxim',
    date: '01/01/2019 10:10:10',
    comments: 'It was very necessary',
  }]);

  const getStore = (nodes: any[] = []) => {
    const onLoad = jest.fn();
    return {
      onLoad,
      store: new Store({
        modules: {
          node: {
            namespaced: true,
            state: () => ({
              nodes,
            }),
            actions: {
              loadNodes: onLoad,
            },
          },
        },
      }),
    };
  };

  const getRouter = (onPushCall) => {
    const onPush = jest.fn((path) => {
      onPushCall(path);
    });
    const RouterMock = jest.fn<VueRouter, any>().mockImplementation(() => ({
      push: onPush,
    } as any));
    return {
      onPush,
      router: new RouterMock(),
    };
  };

  test('should mounts properly', () => {
    const localVue = getLocalVue();
    const { store } = getStore();
    const wrapper = shallowMount(page, {
      localVue,
      store,
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('should renders properly with no data', () => {
    const localVue = getLocalVue();
    const { store } = getStore();
    const wrapper = mount(page, {
      localVue,
      store,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should renders properly with data', () => {
    const localVue = getLocalVue();
    const { store } = getStore(getData());
    const wrapper = mount(page, {
      localVue,
      store,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should call fetch', async () => {
    const localVue = getLocalVue();
    const { store, onLoad } = getStore();
    const wrapper = shallowMount(page, {
      localVue,
      store,
    });
    await wrapper.vm.$options.fetch!(getContext(store));
    expect(onLoad).toBeCalled();
  });

  test('should open page node/1', async () => {
    const localVue = getLocalVue();
    const { store } = getStore(getData());
    let route;
    const { router, onPush } = getRouter(r => route = r);
    const wrapper = mount(page, {
      localVue,
      store,
      mocks: {
        $router: router,
      },
    });
    await wrapper.vm.$options.fetch!(getContext(store));
    const node = wrapper.find('tbody tr.lc-pointer');
    node.trigger('click');
    expect(onPush).toBeCalled();
    expect(route.path).toBe('/node/1');
  });
});
