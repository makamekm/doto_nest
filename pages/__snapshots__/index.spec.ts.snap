// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page index should renders properly with data 1`] = `
<div>
  <div class="hero is-light p-b-6 p-t-6 m-t-7">
    <div class="container p-b-2 p-l-4 p-r-4 p-l-0-desktop p-r-0-desktop" style="width: 100%;">
      <div class="columns is-multiline is-variable is-1 is-vcentered">
        <div class="column">
          <div class="field is-expanded">
            <!---->
            <div class="control has-icons-left is-expanded is-clearfix"><input type="text" autocomplete="on" placeholder="Lookup..." class="input"> <span class="icon is-left"><i class="fab fa-searchengin fa-lg"></i></span>
              <!---->
              <!---->
            </div>
            <!---->
          </div>
        </div>
        <div class="column is-narrow">
          <div class="field">
            <!----> <button class="button is-primary fix-white-on-focus is-expanded"><span class="icon is-small"><i class="fas fa-search"></i></span> <span>Search</span></button>
            <!---->
          </div>
        </div>
        <div class="column is-narrow">
          <div class="has-text-centered"><button type="button" class="button is-transparent is-small fix-white-on-focus is-expanded"><span class="icon is-small"><i class="fas fa-caret-down fas fa-caret-up"></i></span> <span>Filters</span></button></div>
        </div>
        <!---->
      </div>
    </div>
  </div>
  <div class="container m-t-6 p-b-6">
    <div class="columns is-multiline">
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="p-b-2 p-l-4 p-r-4 p-l-0-desktop p-r-0-desktop">
      <nav class="pagination"><a role="button" href="#" disabled="disabled" aria-label="Previous page" class="pagination-previous"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-left mdi-24px"></i></span></a> <a role="button" href="#" aria-label="Next page" class="pagination-next"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a>
        <ul class="pagination-list">
          <!---->
          <!---->
          <li><a role="button" href="#" aria-label="Current page, Page 1." aria-current="true" class="pagination-link is-current">
              1
            </a></li>
          <li><a role="button" href="#" aria-label="Page 2." class="pagination-link">
              2
            </a></li>
          <li><span class="pagination-ellipsis">…</span></li>
          <li><a role="button" href="#" aria-label="Page 5." class="pagination-link">
              5
            </a></li>
        </ul>
        <!---->
      </nav>
    </div>
  </div>
</div>
`;

exports[`page index should renders properly with no data 1`] = `
<div>
  <div class="hero is-light p-b-6 p-t-6 m-t-7">
    <div class="container p-b-2 p-l-4 p-r-4 p-l-0-desktop p-r-0-desktop" style="width: 100%;">
      <div class="columns is-multiline is-variable is-1 is-vcentered">
        <div class="column">
          <div class="field is-expanded">
            <!---->
            <div class="control has-icons-left is-expanded is-clearfix"><input type="text" autocomplete="on" placeholder="Lookup..." class="input"> <span class="icon is-left"><i class="fab fa-searchengin fa-lg"></i></span>
              <!---->
              <!---->
            </div>
            <!---->
          </div>
        </div>
        <div class="column is-narrow">
          <div class="field">
            <!----> <button class="button is-primary fix-white-on-focus is-expanded"><span class="icon is-small"><i class="fas fa-search"></i></span> <span>Search</span></button>
            <!---->
          </div>
        </div>
        <div class="column is-narrow">
          <div class="has-text-centered"><button type="button" class="button is-transparent is-small fix-white-on-focus is-expanded"><span class="icon is-small"><i class="fas fa-caret-down fas fa-caret-up"></i></span> <span>Filters</span></button></div>
        </div>
        <!---->
      </div>
    </div>
  </div>
  <div class="container m-t-6 p-b-6">
    <div class="columns is-multiline">
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
      <div class="column is-12 is-4-desktop">
        <div class="card is-interactable m-t-3 p-t-5 p-b-6 p-l-6 p-r-6">
          <div class="has-text-weight-semibold is-size-6 has-text-font-title">
            Raymond James
          </div>
          <div class="vueperslides no-shadow vueperslides--touchable" aria-label="Slideshow">
            <!---->
            <div class="vueperslides__inner">
              <div class="vueperslides__parallax-wrapper" style="padding-bottom: 66.66666666666666%;" aria-live="polite">
                <div class="vueperslides__track">
                  <div class="vueperslides__track-inner" style="transition-duration: 0ms; transform: translate3d(0%, 0, 0); will-change: auto;">
                    <!---->
                    <div aria-hidden="false" class="vueperslide vueperslide--active vueperslide--visible" style="background-image: url(/uploads/2cr1.jpg);">
                      <!---->
                      <div class="vueperslide__content-wrapper" style="display: none;">
                        <div class="vueperslide__title" style="display: none;">
                          <div></div>
                          <!---->
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
            <!---->
          </div>
          <div>
            <div class="columns is-vcentered m-t-3">
              <div class="column is-size-3 has-text-font-price has-text-grey-dark has-text-weight-semibold has-text-centered">
                500$
              </div>
              <div class="column is-size-7 has-text-weight-semibold has-text-centered"><span class="tag is-light m-t-2 m-b-1">SOFA</span> <span class="tag is-info m-t-2 m-b-1">SALE</span> <span class="tag is-success m-t-2 m-b-1">ECO</span> <span class="tag is-warning m-t-2 m-b-1">LIMITED</span> <span class="tag is-black m-t-2 m-b-1">SOON</span> <span class="tag is-danger m-t-2 m-b-1">OLD</span></div>
            </div>
            <div class="columns m-t-5">
              <div class="column"><button class="button is-transparent fix-white-on-focus is-hero is-expanded"><span class="icon is-small"><i class="fas fa-cart-plus"></i></span> <span>To Cart</span></button></div>
              <div class="column"><button class="button is-hero is-expanded">
                  Overview
                </button></div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="p-b-2 p-l-4 p-r-4 p-l-0-desktop p-r-0-desktop">
      <nav class="pagination"><a role="button" href="#" disabled="disabled" aria-label="Previous page" class="pagination-previous"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-left mdi-24px"></i></span></a> <a role="button" href="#" aria-label="Next page" class="pagination-next"><span class="icon" aria-hidden="true"><i class="mdi mdi-chevron-right mdi-24px"></i></span></a>
        <ul class="pagination-list">
          <!---->
          <!---->
          <li><a role="button" href="#" aria-label="Current page, Page 1." aria-current="true" class="pagination-link is-current">
              1
            </a></li>
          <li><a role="button" href="#" aria-label="Page 2." class="pagination-link">
              2
            </a></li>
          <li><span class="pagination-ellipsis">…</span></li>
          <li><a role="button" href="#" aria-label="Page 5." class="pagination-link">
              5
            </a></li>
        </ul>
        <!---->
      </nav>
    </div>
  </div>
</div>
`;
